### PR TITLE
[Process]DemoFragment DuyLH

### PR DESCRIPTION
**Pull request by duyle112**

### CODE REVIEW

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package vn.asiantech.training;
+
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+
+import java.util.ArrayList;
+
+public class Exfragment extends AppCompatActivity implements EditFragment.OnHeadlineSelectedListener,
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: Incorrect convention. Please, change name Activity

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        InformationFragment.OnHeadlineSelectedListener2, AddFragment.OnFragmentInteractionListener {
+    public static String STUDENT_NAME = "student";
+    public static String POSITION = "position";
+    FragmentManager fragmentManager;
+    ArrayList<Student> students;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_exfragment);
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: change name layout xml for activity

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+
+import java.util.ArrayList;
+
+public class Exfragment extends AppCompatActivity implements EditFragment.OnHeadlineSelectedListener,
+        InformationFragment.OnHeadlineSelectedListener2, AddFragment.OnFragmentInteractionListener {
+    public static String STUDENT_NAME = "student";
+    public static String POSITION = "position";
+    FragmentManager fragmentManager;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    ArrayList<Student> students;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+
+import java.util.ArrayList;
+
+public class Exfragment extends AppCompatActivity implements EditFragment.OnHeadlineSelectedListener,
+        InformationFragment.OnHeadlineSelectedListener2, AddFragment.OnFragmentInteractionListener {
+    public static String STUDENT_NAME = "student";
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    public static String POSITION = "position";
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/Exfragment.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        students = new ArrayList<Student>();
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "21"));
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "22"));
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "23"));
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "24"));
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "25"));
+        students.add(new Student("DUT", "Duy Le", "Da Nang", "26"));
+        ListStudentFragment studentFragment = new ListStudentFragment();
+        fragmentManager = getSupportFragmentManager();
+        studentFragment.passData(students);
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/Exfragment.java]: em chuyển qua dùng bundle, để chuyển dữ liệu qua

File: app/src/main/java/vn/asiantech/training/ListStudentFragment.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ImageButton;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+
+public class ListStudentFragment extends Fragment {
+    ArrayList<Student> students;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/ListStudentFragment.java]: set modifier for variable

File: app/src/main/java/vn/asiantech/training/InformationFragment.java [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+
+public class InformationFragment extends Fragment {
+
+    Student student;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/InformationFragment.java]: set modifier for variables

File: app/src/main/java/vn/asiantech/training/InformationFragment.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                student.setAddress(tvAddress.getText().toString());
+                student.setAge(tvAge.getText().toString());
+                student.setSchool(tvSchool.getText().toString());
+                onButtonPressed(student, position);
+            }
+        });
+        return view;
+    }
+
+    public void onButtonPressed(Student student, int position) {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/InformationFragment.java]: i think needn't create method

File: app/src/main/java/vn/asiantech/training/InformationFragment.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        if (mCallback != null) {
+            mCallback.onFragmentInteraction2(student, position);
+        }
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/InformationFragment.java]: onAttach deprecated, you should being replace onAttach(Context context)

File: app/src/main/java/vn/asiantech/training/InformationFragment.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+        super.onAttach(activity);
+        try {
+            mCallback = (OnHeadlineSelectedListener2) activity;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString() + " must implement OnFragmentInteractionListener2");
+        }
+    }
+
+    public interface OnHeadlineSelectedListener2 {
+        public void onFragmentInteraction2(Student student, int position);
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/InformationFragment.java]: default method into interface always public, so you can delete modifier for method

File: app/src/main/java/vn/asiantech/training/AddFragment.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+            @Override
+            public void onClick(View v) {
+                getActivity().onBackPressed();
+            }
+        });
+        return view;
+    }
+
+    // TODO: Rename method, update argument and hook method into UI event
+    public void onButtonPressed(Student student) {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/AddFragment.java]: i think needn't create method

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.ImageButton;
+
+public class EditFragment extends Fragment {
+    OnHeadlineSelectedListener mCallback;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: set modifier

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+    private Student student;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+    private int position;
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: incorrect convention

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+    public EditFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Bundle bundle = getArguments();
+        if (bundle != null) {
+            student = (Student) bundle.getSerializable("student");
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: should be declare variable for key bundle

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+            position = bundle.getInt("position");
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: should be declare variable for key bundle

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+                student.setAge(edtAge.getText().toString());
+                student.setSchool(edtSchool.getText().toString());
+                onButtonPressed(student, position);
+            }
+        });
+
+        return view;
+    }
+
+    public void onButtonPressed(Student student, int position) {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: i think needn't create method

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+        if (mCallback != null) {
+            mCallback.onFragmentInteraction1(student, position);
+        }
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: onAttach deprecated, you should being replace onAttach(Context context)

File: app/src/main/java/vn/asiantech/training/EditFragment.java [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+        super.onAttach(activity);
+        try {
+            mCallback = (OnHeadlineSelectedListener) activity;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString() + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    public interface OnHeadlineSelectedListener {
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/EditFragment.java]: needn't declare modifier "public" into interface

File: app/src/main/java/vn/asiantech/training/StudentAdapter.java [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+/**
+ * Created by HoangDuy on 14/11/2016.
+ */
+public class StudentAdapter extends ArrayAdapter<Student> {
+
```
Comment (ghost) [inline app/src/main/java/vn/asiantech/training/StudentAdapter.java:16]: em nen add ViewHolder cho listview

File: app/src/main/res/layout/student_row.xml [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+    <TextView
+        android:id="@+id/tvAge"
+        android:layout_width="0dp"
+        android:layout_height="match_parent"
+        android:layout_weight="1"
+        android:gravity="center"
+        android:textColor="@color/colorRed"
+        android:textSize="@dimen/listview_item_fontsize"
+        tools:text="21" />
+</LinearLayout>
```
Comment (ghost) [inline app/src/main/res/layout/student_row.xml:50]: insert break line

**Comment by ghost** [comment]
em xem comment và check lại code nhe Duy

File: app/src/main/java/vn/asiantech/training/Exfragment.java (hunk 7 above)
Comment (duyle112) [reply app/src/main/java/vn/asiantech/training/Exfragment.java]: chỗ này truyền arraylist qua thốn qua anh.Để em nghiên cứu sau.Hôm qua anh Tiến bày cách ni

**Pull request closed without merging**